From d3ec49b564bf8f60ccac9bd7f269009b5ecc415e Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 08 Jun 2021 17:43:54 +0300
Subject: [PATCH] [SW-44814] pci: stop mem_ena while link is down

memory access enable is disabled by default when the pci link is
down so there is no need for the polling.

Change-Id: Iff2aa11b95272285902cfa2dd8994ee6588af83c
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_pci.c b/preboot/module_pci.c
index 8f62a04..3434a7e 100644
--- a/preboot/module_pci.c
+++ b/preboot/module_pci.c
@@ -67,7 +67,9 @@
 	    pci_sts.link_is_up)
 		return FOUND_FAST_RETURN;
 
-	pci_sts.mem_enabled = hl_pci_is_mem_enabled();
+	if (pci_sts.link_is_up)
+		pci_sts.mem_enabled = hl_pci_is_mem_enabled();
+
 	if (pci_sts.mem_enabled != pci_sts.mem_enabled_prev)
 		return FOUND_FAST_RETURN;
 
@@ -108,10 +110,12 @@
 	    pci_sts.link_is_up) {
 		pci_sts.link_is_up = !pci_sts.link_is_up;
 		preboot_puts_raw("PCIe link is ");
-		if (pci_sts.link_is_up)
+		if (pci_sts.link_is_up) {
 			preboot_puts("up\n");
-		else
+		} else {
 			preboot_puts("down\n");
+			pci_sts.mem_enabled = false;
+		}
 	}
 
 	if (pci_sts.mem_enabled_prev != pci_sts.mem_enabled) {
